active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,572 @@
+        />
+```
+
+#### C# (Model)
+```csharp
+    public class Person : ValidationWrapper
+    {
+        [Required]
+        [RegularExpression(@"^[a-zA-Z''-'\s]{1,40}$", ErrorMessage ="Use alphabetical characters (a-Z).")]
+        public string FirstName {
+            get { return GetValue<string>("FirstName"); }
+            set { SetValue("FirstName", value); }
+        }
+    }
+```
+
+## Validation for view-centric developers
+Code a view-centric (Winforms-like) developer would write to enable Validation.
+
+
+### 1.10.1	C# Sample
+#### XAML
+
+```xaml
+<TextBox x:Name="Email" 
+         Header="Email" 
+         Validation.IsRequired="True" 
+         TextChanged="TextBox_TextChanged"/>
+```
+
+#### C#
+```csharp
+private void TextBox_TextChanged(object sender, TextChangedEventArgs e)
+{
+    WinformsValidate(sender, e);
+}
+
+private void WinformsValidate(object sender, TextChangedEventArgs e)
+{
+    var foo = sender as TextBox;
+    Validation.ClearErrors(foo);
+    Validation.AddErrorHandler(foo, OnValidationError);
+
+    if (!new EmailAddressAttribute().IsValid(foo.Text))
+    {
+
+        var ErrorsCollection = Validation.GetErrors(foo);
+        ErrorsCollection.Add(new ValidationError("Email is not a valid email address"));
+    }
+}
+
+private void OnValidationError(object sender, ValidationErrorEventArgs e)
+{
+    if (e.Action == ValidationErrorEventAction.Added)
+        // Do something when an error message was added
+    else
+        // Do something when an error was removed.
+
+        SaveButton.IsEnabled = Validation.GetErrors(e.Source);
+}
+```
+
+<!-- Use this section to explain the features of the API, showing
+example code with each description. The general format is: 
+  feature explanation,
+  example code
+  feature explanation,
+  example code
+  etc.-->
+  
+<!-- Code samples should be in C# and/or C++/WinRT -->
+
+<!-- As an example of this section, see the Examples section for the PasswordBox control 
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+
+
+# Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples section. -->
+
+<!-- APIs should only throw exceptions in exceptional conditions; basically,
+only when there's a bug in the caller, such as argument exception.  But if for some
+reason it's necessary for a caller to catch an exception from an API, call that
+out with an explanation either here or in the Examples -->
+
+# API Notes
+<!-- Option 1: Give a one or two line description of each API (type
+and member), or at least the ones that aren't obvious
+from their name.  These descriptions are what show up
+in IntelliSense. For properties, specify the default value of the property if it
+isn't the type's default (for example an int-typed property that doesn't default to zero.) -->
+
+<!-- Option 2: Put these descriptions in the below API Details section,
+with a "///" comment above the member or type. -->
+
+# API Details
```

Comment:
Almost none of this API is demonstrated or explained in the Examples

Reply:
We are updating to match the latest IDL provided and clean up the samples.